sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/CheckpointContext.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.WebJobs.EventHubs
+{
+    public class CheckpointContext
```

Comment:
Happy to implement it this way, although as per `Considerations` in the PR description, would this not be a breaking change due to the public API currently binding on `PartitionContext` rather than `EventProcessorHostPartitionContext`.

I assume either binding would have to be supported to ensure that isn't the case?

Thanks for the quick feedback so far.

Reply:
Based on this proposal I've implemented a `TriggerPartitionContext` (in case there are more properties later which are not checkpoint specific) and pushed it.

If this is agreeable, we may need to make a tweak to the unit test mocks to support it.

There's an issue now the context is referenced during tests where the `ProcessorHost` is null now the `PartitionContext` property is required during the test (when the checkpoint property is updated).

In short this is currently being worked around in the constructor of `TriggerPartitionContext` with defaulting of values but I didn't want to go further than that without API approval.